client/src/app/pages/applications/applications-table/components/column-application-name.tsx

```diff
@@ -0,0 +1,178 @@
+import { Table, Tbody, Td, Thead, Tr } from "@patternfly/react-table";
+import { Task } from "@app/api/models";
+
+type ApplicationStatus =
+  | "None"
+  | "Running"
+  | "Queued"
+  | "Failed"
+  | "Canceled"
+  | "Success";
+
+interface StatusData {
+  popoverVariant: PopoverProps["alertSeverityVariant"];
+  icon: React.FunctionComponent;
+  headerText: string;
+}
+
+/**
+ * Step through the task data of an application and return its summarized task status.
+ */
+const chooseApplicationTaskStatus = ({
+  tasks,
+}: DecoratedApplication): ApplicationStatus => {
+  return !tasks.exist
+    ? "None"
+    : tasks.latestHasRunning
+    ? "Running"
+    : tasks.latestHasQueued
+    ? "Queued"
+    : tasks.latestHasFailed
+    ? "Failed"
+    : tasks.latestHasCanceled
+    ? "Canceled"
+    : "Success";
+};
+
+/* TODO: This can be better aligned with `TaskStateIcon` in future. */
+const statusMap: Record<ApplicationStatus, StatusData> = {
+  None: {
+    popoverVariant: undefined,
+    headerText: "This application has no tasks",
+    icon: () => <TimesCircleIcon />,
+  },
+  Running: {
+    popoverVariant: "info",
+    headerText: "One or more tasks are running",
+    icon: () => (
+      <Icon status="info">
+        <InProgressIcon />
+      </Icon>
+    ),
+  },
+  Queued: {
+    popoverVariant: "info",
+    headerText: "One or more tasks are queued to run",
+    icon: () => (
+      <Icon status="info">
+        <PendingIcon />
+      </Icon>
+    ),
+  },
+  Failed: {
+    popoverVariant: "danger",
+    headerText: "One or more tasks failed",
+    icon: () => (
+      <Icon status="danger">
+        <ExclamationCircleIcon />
+      </Icon>
+    ),
+  },
+  Canceled: {
+    popoverVariant: "info",
+    headerText: "One of more tasks were canceled",
+    icon: () => (
+      <Icon status="info">
+        <TimesCircleIcon />
+      </Icon>
+    ),
+  },
+  Success: {
+    popoverVariant: "success",
+    headerText: "All tasks succeeded",
+    icon: () => (
+      <Icon status="success">
+        <CheckCircleIcon />
+      </Icon>
+    ),
+  },
+};
+
+const linkToTasks = (applicationName: string) => {
+  const search = `t:filters={"application":["${applicationName}"]}`;
+  return `${formatPath(Paths.tasks, {})}?${search}`;
+};
+
+const linkToDetails = (task: Task) => {
+  return formatPath(Paths.taskDetails, {
+    taskId: task.id,
+  });
+};
+
+export const ColumnApplicationName: React.FC<{
+  application: DecoratedApplication;
+}> = ({ application }) => {
+  const status = statusMap[chooseApplicationTaskStatus(application)];
+  const StatusIcon = status.icon;
+
+  const bodyContent = (
+    <Table variant="compact" borders={false}>
+      <Thead>
+        <Tr>
+          <Td>Id</Td>
+          <Td>Kind</Td>
+          <Td>Started</Td>
+        </Tr>
+      </Thead>
+      <Tbody>
+        {Object.entries(application.tasksByKind)
+          .sort((a, b) => universalComparator(a[0], b[0]))
+          .map(([kind, [task]]) => {
```

Comment:
This relies on the tasks array in the `tasksByKind` to be sorted by `createTime` descending.  The fetch hook does that for us (since `createTime` is not a sortable field on the API).  The first entry in the array is always the most recently created task for that kind.

Reply:
it's the most recently created but necessarily started. in this sense it might be not the most recent.

Reply:
ok. create date seems sufficient.